src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -346,6 +382,79 @@ internal SelectedPropertiesNode SelectedProperties
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value that indicates whether the writer should put key values in their own URI segment when automatically building URIs.
+        /// If this value is false, automatically-generated URLs will take the form "../EntitySet('KeyValue')/..".
+        /// If this value is true, automatically-generated URLs will take the form "../EntitySet/KeyValue/..".
+        /// This setting only applies to URLs that are automatically generated by the <see cref="ODataMessageWriter" /> and does not modify URLs explicitly provided by the user.
+        /// </summary>
+        public bool EnableWritingKeyAsSegment { get; set; }
+
+        /// <summary>
+        /// Get whether to write OData control information without a prefix
+        /// True if control information can be read without prefix 'odata.', otherwise false.
+        /// The default value is false for OData 4.0 and true for OData 4.01.
+        /// The option is applied during deserialization.
+        /// </summary>
+        /// <returns>Whether to omit the OData prefix for the specified version.</returns>
+        public bool GetOmitODataPrefix()
+        {
+            return this.enableWritingODataAnnotationWithoutPrefix;
+        }
+
+        /// <summary>
+        /// Version-specific behavior for writing OData control information without a prefix
+        /// True if control information can be read without prefix 'odata.', otherwise false.

```

Comment:
You're mixing up "reading" and "writing" in the doc comments

Reply:
I copied the doc comments form the simplified options. I think it had them mixed up as well. Let me fix that.